docker/Dockerfile

```diff
@@ -205,22 +218,29 @@ RUN echo "Building Avalon SDK\n" \
 
 # Build Final image and install dependent modules
 FROM base_image as final_image
-
-RUN apt-get update \
- && apt-get install -y -q python3-pip git \
- && pip3 install json-rpc web3 py-solc-x nose2 \
- && echo "Install fabric python sdk\n" \
- && pip3 install virtualenv \
- && echo "fastapi==0.27.*" >>requirement.txt \
- && echo "grpcio-tools==1.21.*" >>requirement.txt \
- && echo "uvicorn==0.7.*" >>requirement.txt \
- && pip3 install -r requirement.txt \
- && pip3 install git+https://github.com/hyperledger/fabric-sdk-py.git \
-# nest_asyncio allows to run nested asyncio event loops
- && pip3 install nest_asyncio \
- && apt-get autoremove --purge -y -q python3-pip git
- 
-# Install specific version of solc to support versions <0.6.0
+ARG DISTRO
+COPY ./scripts/install_packages /project/avalon/scripts/
+
+RUN if [ "$DISTRO" = "centos" ] ; then \
+      packages="openssl-devel wget make gcc"; \
+      /project/avalon/scripts/install_packages -c install -q "$packages"; \
+      cd /tmp \
+      && wget https://www.python.org/ftp/python/3.6.9/Python-3.6.9.tgz \
+      && tar -xvzf Python-3.6.9.tgz \
+      && cd Python-3.6.9 \
+      && ./configure \
+      && make \
+      && make install; \
+    fi
+
+RUN packages="python3-pip git"; \
```

Comment:
Not related to tis PR. But it makes more sense to keep a separate Shell Dockerfile for direct mode and blockchain.